### DISCUSSION TITLE
AV1 Encoder

### DISCUSSION THREAD

**Discussion by SliWillY** [LizardByte/community#245] [Sunshine] [answered]
hi everyone,

since new graphic cards support AV1 encoding are we getting it any time soon?

**Comment by cgutman** [comment]
AV1 support is now in [Sunshine Nightly](https://github.com/LizardByte/Sunshine/releases/tag/nightly-dev). It is currently only supported for NVENC on Windows until we update FFmpeg.

Nightly versions of [Moonlight Qt](https://ci.appveyor.com/project/cgutman/moonlight-qt) and Moonlight Android support requesting AV1 as an experimental opt-in feature.

**Reply by YinsenW** [reply]
How can I download the  nightly build of moonlight ? Thank you so much

**Reply by cgutman** [reply]
Moonlight Qt has nightly builds available. If you click on the link I posted in my previous post, you'll click on the desired job (Windows 1, Mac is 2, Steam Link is 3, and Linux AppImage is 4), then click on the Artifacts tab and the downloads will be there.

**Reply by YinsenW** [reply]
thanks you，i found the mac build. by the way i need the android nightly version,how can i download it ?

**Comment by apoklyps3** [comment]
my current GPU doesn't support AV1 encoding and I'm planning on upgrading soon.
I see AV1 support has been added to sunshine and moonlight-qt
My current contenders would be Nvidia RTX 4070 or AMD RX 7800XT.
I keep reading online the Nvidia's AV1 encoder is way better than AMD's. Is that true?
So AMD cards are not yet supported or tested for AV1 gamestreaming.

**Reply by cgutman** [reply]
AMD's encoder is not supported by Sunshine yet, but it will be once FFmpeg 6.1 is released and we upgrade our FFmpeg version. I have tested it with a custom build of FFmpeg, so we know it works.

I would lean toward Nvidia still. Even ignoring any quality differences (which I didn't really notice), NVENC supports reference frame invalidation which allows much better performance and image quality when recovering after a lost frame.

**Reply by VoidTallica** [reply]
Does the RX6800 support AV1? I read some news that the 6000x series would support it, but i'm not sure of it. Hope i can finally get the performance i want to use sunshine in this new update.

**Reply by ReenigneArcher** [reply]
AV1 DECODE only according to https://www.amd.com/en/products/graphics/amd-radeon-rx-6800

**Comment by apoklyps3** [comment]
Too bad nvidia is still the better option for encoding.
7800xt seems to offer a better value at a lower price when compared to RTX 4070.
How bad is missing the " reference frame invalidation " feature and is it a hardware perk or something that could be added down the line on AMD GPU. Also is it something that only affects AV1?

**Reply by bigdrew** [reply]
I actually just replaced a 3060 with a 7800xt.  It's my first discreet AMD board and I'm really happy with it.  The latency using Sunshine is at least as low as the NVIDIA card and the gamestream quality in terms of stability and artifacts is just as excellent.  I think it actually looks a bit better subjectively because it has a slight sharpness to it, and I don't know why because I didn't have anything set at the driver level for either board that should've affected this.  I obviously can't test AV1 yet for streaming but the recordings look great even at low bitrates.  I was a little reluctant to switch away from NVIDIA since gamestreaming performance is priority for me personally but I'm quite pleased with the new board and relieved I didn't sacrifice anything by switching.

**Comment by ivdimitro** [comment]
Hi, any improvement in quality/latency with the AV1? I think the most important thing will be to reduce the latency as the quality seems good enough even with H.265

**Comment by cgutman** [comment, marked as answer]
AV1 encoding is now supported for all supported GPUs and OSes in nightly builds.

[a comment by an automated account is not shown]

**Comment by lastmetroid** [comment]
Whats the current status on this, is AV1 usable or are H264 and H265 still better because of lower latency both to encode and decode?

**Comment by CarloApri** [comment]
Can we encode AV1 with the software encoder? My moonlight host fallsback to H265 for me. Both devices updated to latest stable build as of now.

**Reply by cgutman** [reply]
No, AV1 is not supported for software encoding due to limitations of the SVT-AV1 encoder. I have filed a feature request to address the issue that's blocking us: https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/2120

There is also the matter of very poor performance. Even my Ryzen 9 7950X with AVX-512 support isn't powerful enough to deliver a playable frame rate using the low latency presets (which eliminates the possibility of frame-level multithreading). Even if it did work, the performance would probably be so bad that you couldn't actually use it for anything other than very low resolutions (below 1080p).

**Comment by lastmetroid** [comment]
I tested AV1 with Moonlight and Sunshine and it is kinda unplayble compared to H264 or H265. There is too high latency. Tested with RTX4080 as hw encoder on host and an Intel N100 as client with AV1 hw decode. Really disappointing. There is a really large latency and mouse lags behind noticable.

**Reply by cgutman** [reply]
Your client is probably the issue. I use AV1 for streaming on many of my client systems (i7 1280P iGPU, 13900H iGPU, Arc A770, RTX 3050 mobile, Pixel 6 Pro) and latency is comparable to HEVC both for HDR and SDR usage.

**Reply by lastmetroid** [reply]
What resolution and what frame rate do you use it with? See the issue here: https://github.com/moonlight-stream/moonlight-qt/issues/1296

Is there a possibility Sunshine implements a forced FPS setting?
